### PR TITLE
[fix] types

### PR DESCRIPTION
**Pull request by cymarechal**

### CODE REVIEW

File: min_queue.c [hunk 1]
```diff
@@ -59,8 +59,10 @@ struct minqueue {
 minqueue_t *minq_new(minq_key_cmp cmp)
 {
 	minqueue_t *q;
-	if (!cmp || !(q = malloc(sizeof(*q))))
+	if (!cmp || !(q = malloc(sizeof(*q)))) {
+		free(q);
```
Comment (qdeconinck) [inline min_queue.c]: I don't think this change is right. Either there is no cmp (and then no malloc is done) or the malloc fails (and no free is required)

**Review by qdeconinck — COMMENTED** [review]
Ok for the types, less for the change in the queue

File: min_queue.c (hunk 1 above)
Comment (cymarechal) [reply min_queue.c]: Good point. Actually I found it with [cppcheck](http://cppcheck.sourceforge.net), who clearly indicates a memory leak here. Double checking it here proves that we indeed do not need to free memory. Thank you for pointing that out.

Comment (qdeconinck) [reply min_queue.c]: Static analyzers sometimes produce false positives :-)

**Pull request merged**